src/imp/schannel.rs

```diff
@@ -140,15 +154,126 @@ impl Identity {
+                } 
+
+                let identity = match identity {
+                    Some(identity) => identity,
+                    None => {
+                        return Err(io::Error::new(io::ErrorKind::InvalidInput,"No identity found in provided store").into());
+                    }
+                };
+                return Ok(Identity { cert: identity })
+            }
+
+            OsProviderParameters::ContextFromFile {file_path} => {
+                let store = CertStore::open_file(&file_path.as_path())?;
+                // set identity to the cert that matches the key inside the pfx file
+                let mut identity = None;
+                for cert in store.certs() {
+                    if cert
+                        .private_key()
+                        .silent(true)
+                        .compare_key(true)
+                        .acquire()
+                        .is_ok()
+                    {
+                        identity = Some(cert);
+                        break;
+                    }
+                }
+                let identity = match identity {
+                    Some(identity) => identity,
+                    None => {
+                        return Err(io::Error::new(io::ErrorKind::InvalidInput,"No identity found in provided file").into());
+                    }
+                };
+                return Ok(Identity { cert: identity })
+            }
+        }
+    }
 }
 
+
 // The name of the container must be unique to have multiple active keys.
 fn gen_container_name() -> String {
     use std::sync::atomic::{AtomicUsize, Ordering};
     static COUNTER: AtomicUsize = AtomicUsize::new(0);
     format!("native-tls-{}", COUNTER.fetch_add(1, Ordering::Relaxed))
 }
 
+fn parse_engine_string(engine_string: &OsStr) -> io::Result<OsProviderParameters> {         
+    let converted_str = match engine_string.to_str() {
+        Some(s) => s,
+        None => return Err(io::Error::new(io::ErrorKind::InvalidInput,"Contains invalid UTF-8 characters").into()),
+    };
+
+    if let Some((first, second)) = converted_str.split_once(":") {
+        let first = first.trim();
+        if first == "file" {
+            let path = PathBuf::from(second.trim());
+            let context_from_file = OsProviderParameters::ContextFromFile { file_path: path };  
+            return Ok(context_from_file); 
+        } 
+
+        if first == "user" || first == "machine" {
+            let parts: Vec<&str> = second.split(':').map(str::trim).collect();
+            if parts.len() != 2 {
+                return Err(io::Error::new(io::ErrorKind::InvalidInput,"Missing store name and/or certificate thumbprint").into())
+            }            
+
+            let decoded_hex = hex::decode(parts[1].to_string())
+            .map_err(|_| io::Error::new(io::ErrorKind::InvalidInput,"Hex decode failed"))?;
+            
+            let context_from_store = OsProviderParameters::ContextFromStore {
```

Comment:
```suggestion
            Ok(OsProviderParameters::ContextFromStore {
```

and fixup the end of the statement/return statement